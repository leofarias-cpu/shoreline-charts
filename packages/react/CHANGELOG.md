# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
