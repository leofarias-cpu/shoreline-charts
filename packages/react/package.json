{
  "name": "@vtex/admin-ui-react",
  "license": "MIT",
  "version": "0.0.0",
  "main": "dist/index.js",
  "module": "dist/admin-ui-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/onda-core": "^0.1.0-labs.8",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "@vtex/onda-util": "^0.1.0-labs.3",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
