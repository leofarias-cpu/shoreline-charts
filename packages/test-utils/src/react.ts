import {
  render,
  renderHook,
  act,
  screen,
  fireEvent,
} from '@testing-library/react'

export { render, renderHook, act, screen, fireEvent }
