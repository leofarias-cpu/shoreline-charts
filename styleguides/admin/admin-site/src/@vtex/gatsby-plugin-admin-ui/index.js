export default {
  sizes: {
    header: '4.5rem',
    content: '90rem',
  },
  space: {
    content: '4.5rem',
  },
}
