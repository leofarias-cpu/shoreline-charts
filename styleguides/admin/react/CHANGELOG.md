# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.4.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.2...@vtex/onda-react@0.4.3) (2021-07-20)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.1...@vtex/onda-react@0.4.2) (2021-07-19)

**Note:** Version bump only for package @vtex/onda-react





## [0.4.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.4.0...@vtex/onda-react@0.4.1) (2021-07-16)


### Bug Fixes

* **onda-react:** remove the cleaning of options properties from jsx ([b8ec31c](https://github.com/vtex/onda/commit/b8ec31cf1e557996ea8eb7027edbcfcbe55845fb))





# [0.4.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.3.0...@vtex/onda-react@0.4.0) (2021-07-15)


### Features

* **admin-ui/ultimate-table:** support sorting ([7453750](https://github.com/vtex/onda/commit/745375001bd28bb73e8a229f14c12e78bd0385da))





# [0.3.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.2...@vtex/onda-react@0.3.0) (2021-07-15)


### Bug Fixes

* **onda-react:** jsx refs type inference ([5a24609](https://github.com/vtex/onda/commit/5a2460960c419a272f16d76a43f9575247643f2a))
* **onda-react:** jsx rendering issues ([6fbe2b5](https://github.com/vtex/onda/commit/6fbe2b585ab083a99a1aa806de3ed0e23ac2451f))
* **onda-react:** typescript inference for jsx.elements ([fb9e365](https://github.com/vtex/onda/commit/fb9e365df6a877a274bc3ea26131a4015fbbcd33))


### Features

* **onda-react:** add ComponentProps type ([e13433c](https://github.com/vtex/onda/commit/e13433c2b9611de3cd6723541fe82064bd1f10cb))





## [0.2.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.1...@vtex/onda-react@0.2.2) (2021-07-05)

**Note:** Version bump only for package @vtex/onda-react





## [0.2.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.2.0...@vtex/onda-react@0.2.1) (2021-07-01)


### Bug Fixes

* **onda-core:** build imports ([9d30f9e](https://github.com/vtex/onda/commit/9d30f9e743d75b03995547e4d3052aa5a510e095))





# [0.2.0](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.1...@vtex/onda-react@0.2.0) (2021-07-01)


### Bug Fixes

* **onda-react:** remove unnecessary state ([1465e69](https://github.com/vtex/onda/commit/1465e695a6cb9be6ba3773d0acf51194a0713ea2))
* **onda-react:** resolve jsx hof types ([2e185ac](https://github.com/vtex/onda/commit/2e185ac2dd319f30d1eb4332a13a2299a38e06cb))


### Features

* **onda-react:** rephrase createComponent as jsx ([dd157ff](https://github.com/vtex/onda/commit/dd157ffa2500ed8e911b7d2d04db65268fc7f27b))





## [0.1.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.4...@vtex/onda-react@0.1.1) (2021-06-29)


### Bug Fixes

* **onda-react:** package.json version ([fb98ac4](https://github.com/vtex/onda/commit/fb98ac411f80ae253225940a7c687cb458677e4a))





# 0.1.0-labs.4 (2021-06-24)


### Features

* **onda-packages:** add onda packages ([b619064](https://github.com/vtex/onda/commit/b619064d04d190d5615a2832dbcb86d96efd16ee))





# [0.1.0-labs.3](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.2...@vtex/onda-react@0.1.0-labs.3) (2021-05-19)


### Features

* **onda:** allow global styles from theme ([aff3fa4](https://github.com/vtex/onda/commit/aff3fa449c6b92004312bf8ee4f648d7e6258dc0))
* **onda-core:** transform theme values on css variables ([3a8a379](https://github.com/vtex/onda/commit/3a8a379c3ccebf4738b65d39fc38aa8ac594e621))
* **onda-runtime-emotion:** allow default entries ([73a0dea](https://github.com/vtex/onda/commit/73a0deafda0b18ca09ca6eca00637d0c976055d8))





# [0.1.0-labs.2](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.1...@vtex/onda-react@0.1.0-labs.2) (2021-05-13)


### Features

* **onda-react:** allow state property ([47bfb26](https://github.com/vtex/onda/commit/47bfb26bdb2f2f5496a9458505913f8133dfbee2))





# [0.1.0-labs.1](https://github.com/vtex/onda/compare/@vtex/onda-react@0.1.0-labs.0...@vtex/onda-react@0.1.0-labs.1) (2021-05-13)

**Note:** Version bump only for package @vtex/onda-react





# 0.1.0-labs.0 (2021-05-13)


### Features

* **onda:** add /react package ([4728b63](https://github.com/vtex/onda/commit/4728b631011945c4bd25a6e2de57c2d2e4f52c17))
