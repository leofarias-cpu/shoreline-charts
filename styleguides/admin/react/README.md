# Onda component model

## jsx

You can use the `jsx` function to create new components using elements or components as reference. It generates an `OndaJSXComponent` that inherits and enhances the reference types and styles.

### Referencing elements

To reference elements, simply pass its string representation to the function.

```jsx
const BlueSquare = jsx('div')({
  bg: 'blue',
  size: 50,
})

render(<BlueSquare />)
```

#### Aliasing

`jsx` is aliased with any valid `JSX.Element` for faster typing.

```jsx
const BlueSquare = jsx.div({
  bg: 'blue',
  size: 50,
})

render(<BlueSquare />)
```

### Referencing components

The idea is the same as elements. It will inherit the types and behavior of the passed component.

```jsx
import { motion } from 'framer-motion'

const Scale = jsx(motion.div)({
  bg: 'blue',
  size: 50,
})

render(<Scale animate={{ scale: 2 }} transition={{ duration: 0.5 }} />)
```
