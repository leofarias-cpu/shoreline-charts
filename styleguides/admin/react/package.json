{
  "name": "@vtex/onda-react",
  "license": "MIT",
  "version": "0.1.0",
  "main": "dist/index.js",
  "module": "dist/onda-react.esm.js",
  "typings": "dist/index.d.ts",
  "publishConfig": {
    "access": "public",
    "registry": "https://registry.npmjs.org"
  },
  "files": [
    "dist",
    "src"
  ],
  "engines": {
    "node": ">=10"
  },
  "repository": {
    "directory": "packages/react",
    "type": "git",
    "url": "git+https://github.com/vtex/onda.git"
  },
  "bugs": {
    "url": "https://github.com/vtex/onda/issues"
  },
  "scripts": {
    "start": "tsdx watch --verbose --noClean",
    "build": "tsdx build --tsconfig tsconfig.json",
    "build:fast": "tsdx build --tsconfig tsconfig.json --transpileOnly --format esm,cjs",
    "test": "tsdx test --passWithNoTests",
    "storybook": "start-storybook -p 6005",
    "storybook:build": "yarn --cwd ../../ build && yarn run build-storybook"
  },
  "peerDependencies": {
    "@emotion/css": ">=11",
    "@emotion/react": ">=11",
    "react": ">=16",
    "react-dom": ">=16"
  },
  "devDependencies": {
    "@vtex/admin-core": "^0.10.10",
    "@vtex/onda-util": "^0.1.0-labs.2",
    "polished": "^4.1.2"
  },
  "dependencies": {
    "@emotion/is-prop-valid": "^1.1.0",
    "react-is": "^17.0.2",
    "tiny-invariant": "^1.1.0",
    "tiny-warning": "^1.0.3"
  }
}
