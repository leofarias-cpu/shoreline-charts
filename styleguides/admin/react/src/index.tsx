export * from './tag'
export * from './domElements'
export * from './jsx'
export * from './types'
export * from './useStyleSheet'
export * from './util'
export * from './symbols'
