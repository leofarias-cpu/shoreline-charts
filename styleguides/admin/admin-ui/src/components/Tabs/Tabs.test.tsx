import React from 'react'
import { render } from '@testing-library/react'

import { Tabs, useTabState, TabStateReturn } from './index'
import { ThemeProvider } from '@vtex/admin-core'
import { axe } from 'jest-axe'

function TabsState({
  children,
}: {
  children: (state: TabStateReturn) => JSX.Element
}) {
  const state = useTabState({ baseId: 'tabs-test' })

  return children(state)
}

describe('Tabs tests', () => {
  it('should have overridable styles', () => {
    const { getByTestId } = render(
      <ThemeProvider>
        <TabsState>
          {(state) => (
            <Tabs state={state}>
              <Tabs.List
                aria-label="overridable-tabs"
                data-testid="tabs-list"
                styleOverrides={{ bg: 'azure' }}
              >
                <Tabs.Tab
                  data-testid="tabs-tab"
                  styleOverrides={{ bg: 'azure' }}
                  label="Tab 1"
                  id="1"
                />
              </Tabs.List>
              <Tabs.Content
                data-testid="tabs-content"
                id="1"
                styleOverrides={{ bg: 'azure' }}
              >
                Tab 1 Content
              </Tabs.Content>
            </Tabs>
          )}
        </TabsState>
      </ThemeProvider>
    )

    expect(getByTestId('tabs-list')).toHaveStyleRule(
      'background-color',
      'azure'
    )
    expect(getByTestId('tabs-tab')).toHaveStyleRule('background-color', 'azure')
    expect(getByTestId('tabs-content')).toHaveStyleRule(
      'background-color',
      'azure'
    )
  })

  it('should match snapshot', () => {
    const { asFragment } = render(
      <ThemeProvider>
        <TabsState>
          {(state) => (
            <Tabs state={state}>
              <Tabs.List aria-label="fixed-tabs">
                <Tabs.Tab label="Tab 1" id="1" />
                <Tabs.Tab label="Tab 2" id="2" />
              </Tabs.List>
              <Tabs.Content id="1">Tab 1 Content</Tabs.Content>
              <Tabs.Content id="2">Tab 2 Content</Tabs.Content>
            </Tabs>
          )}
        </TabsState>
        <TabsState>
          {(state) => (
            <Tabs state={state}>
              <Tabs.List fluid aria-label="fluid-tabs">
                <Tabs.Tab label="Tab 1" id="1" />
                <Tabs.Tab label="Tab 2" id="2" />
              </Tabs.List>
              <Tabs.Content id="1">Tab 1 Content</Tabs.Content>
              <Tabs.Content id="2">Tab 2 Content</Tabs.Content>
            </Tabs>
          )}
        </TabsState>
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  it('should not have any violations', async () => {
    const { container } = render(
      <ThemeProvider>
        <TabsState>
          {(state) => (
            <Tabs state={state}>
              <Tabs.List fluid aria-label="tabs">
                <Tabs.Tab label="Tab 1" id="1" />
                <Tabs.Tab label="Tab 2" id="2" />
              </Tabs.List>
              <Tabs.Content id="1">Tab 1 Content</Tabs.Content>
              <Tabs.Content id="2">Tab 2 Content</Tabs.Content>
            </Tabs>
          )}
        </TabsState>
      </ThemeProvider>
    )

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
})
