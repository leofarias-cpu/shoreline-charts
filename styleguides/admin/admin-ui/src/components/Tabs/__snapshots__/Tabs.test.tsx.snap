// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs tests should match snapshot 1`] = `
<DocumentFragment>
  .emotion-2 {
  padding-left: 1rem;
  padding-right: 1rem;
  width: 100%;
  display: inline-block;
}

.emotion-1 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-style: solid;
  border-bottom-color: transparent;
  border-bottom-width: 2px;
  border-top-style: solid;
  border-top-color: transparent;
  border-top-width: 2px;
  height: 44px;
  min-width: 110px;
  cursor: pointer;
  position: relative;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  text-transform: uppercase;
  background-color: transparent;
  color: #707685;
}

.emotion-1:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-1:hover {
  color: #214491;
}

.emotion-0 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-style: solid;
  border-bottom-color: #2953B2;
  border-bottom-width: 2px;
  border-top-style: solid;
  border-top-color: transparent;
  border-top-width: 2px;
  height: 44px;
  min-width: 110px;
  cursor: pointer;
  position: relative;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  text-transform: uppercase;
  background-color: transparent;
  color: #2953B2;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-label="fixed-tabs"
    class="emotion-2"
    id="tabs-test"
    role="tablist"
  >
    <button
      aria-controls="tabs-test-3"
      aria-label="Tab 1"
      aria-selected="true"
      class="emotion-0"
      id="1"
      role="tab"
      tabindex="0"
    >
      Tab 1
    </button>
    <button
      aria-controls="tabs-test-4"
      aria-label="Tab 2"
      aria-selected="false"
      class="emotion-1"
      id="2"
      role="tab"
      tabindex="-1"
    >
      Tab 2
    </button>
  </div>
  

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-labelledby="1"
    class="emotion-0"
    id="tabs-test-3"
    role="tabpanel"
    style=""
    tabindex="0"
  >
    Tab 1 Content
  </div>
  

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-labelledby="2"
    class="emotion-0"
    hidden=""
    id="tabs-test-4"
    role="tabpanel"
    style="display: none;"
    tabindex="0"
  >
    Tab 2 Content
  </div>
  .emotion-1 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-style: solid;
  border-bottom-color: transparent;
  border-bottom-width: 2px;
  border-top-style: solid;
  border-top-color: transparent;
  border-top-width: 2px;
  height: 44px;
  min-width: 110px;
  cursor: pointer;
  position: relative;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  text-transform: uppercase;
  background-color: transparent;
  color: #707685;
}

.emotion-1:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-1:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

.emotion-1:hover {
  color: #214491;
}

.emotion-2 {
  padding-left: 1rem;
  padding-right: 1rem;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: space-evenly;
  -ms-flex-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  justify-content: space-evenly;
}

.emotion-2 >button {
  width: 100%;
}

.emotion-0 {
  font-variation-settings: 'wght' 92;
  border: none;
  border-radius: 4px;
  border-bottom-left-radius: 0px;
  border-bottom-right-radius: 0px;
  border-bottom-style: solid;
  border-bottom-color: #2953B2;
  border-bottom-width: 2px;
  border-top-style: solid;
  border-top-color: transparent;
  border-top-width: 2px;
  height: 44px;
  min-width: 110px;
  cursor: pointer;
  position: relative;
  padding-left: 1.5rem;
  padding-right: 1.5rem;
  text-transform: uppercase;
  background-color: transparent;
  color: #2953B2;
}

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-label="fluid-tabs"
    class="emotion-2"
    id="tabs-test"
    role="tablist"
  >
    <button
      aria-controls="tabs-test-3"
      aria-label="Tab 1"
      aria-selected="true"
      class="emotion-0"
      id="1"
      role="tab"
      tabindex="0"
    >
      Tab 1
    </button>
    <button
      aria-controls="tabs-test-4"
      aria-label="Tab 2"
      aria-selected="false"
      class="emotion-1"
      id="2"
      role="tab"
      tabindex="-1"
    >
      Tab 2
    </button>
  </div>
  

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-labelledby="1"
    class="emotion-0"
    id="tabs-test-3"
    role="tabpanel"
    style=""
    tabindex="0"
  >
    Tab 1 Content
  </div>
  

.emotion-0:focus:not([data-focus-visible-added]) {
  outline: none;
  box-shadow: none;
}

.emotion-0:focus {
  outline: none;
  box-shadow: 0rem 0rem 0rem 0.125rem #8DB6FA;
}

<div
    aria-labelledby="2"
    class="emotion-0"
    hidden=""
    id="tabs-test-4"
    role="tabpanel"
    style="display: none;"
    tabindex="0"
  >
    Tab 2 Content
  </div>
</DocumentFragment>
`;
